docs/DeveloperGuide.md

```diff
@@ -1,34 +1,330 @@
+Diagrams found in our documentation were generated using <a href="https://plantuml.com/">PlantUML</a> and references were made to <a href="https://github.com/se-edu/addressbook-level2/tree/master/src/seedu/addressbook">addressbook-level2</a> for the structure of the classes and packages. The structures have been modified to meet the needs of our application.
+
+#### <a id="overview"><ins>2.1 Architecture Overview</ins></a>
+
+<p align="center">
+  <img alt="NotUS Architecture Overview" src="diagrams/out/Architecture_Overview.png" />
+  <br><em>Figure 1</em>
+</p>
+
+The architecture design is given in the diagram above. The main components of NotUS are:
+
+1. `InterfaceManager`: Manages the user input as well as the message output from application.
+1. `ParserManager`: Creates a suitable parser, based on the command, to make sense of user message. The respective parsers then make sense of the information and calls the respective commands.
+1. `Command`: Executes the necessary tasks, depending on the respective command calls.
+1. `TagManager`: Stores and manages the creation and deletion of tags and other tag-related functionality.
+1. `Timetable`: Stores and manages the creation and deletion of events and other event-related functionality.
+1. `Notebook`: Stores and manages the creation and deletion of notes and other note-related functionality.
+1. `StorageManager`: Manages the loading of existing saved files and exporting of data to human-editable files.
+
+A Program Evaluation Review Technique (PERT) Chart was created prior to the start of developing NotUS and was constantly updated based on progress and updates from the development team's weekly meetings. A PERT chart is a project management tool that provides a visual representation of a project's timeline. The chart breaks down the individual tasks and aids in identifying task dependencies. A diagram of the PERT chart used for this application is shown below.
+
+<p align="center">
+  <img alt="PERT Chart" src="diagrams/out/PERT_Chart.png" />
+  <br><em>Figure 2</em>
+</p>
+
+#### <a id="notus"><ins>2.2 NotUS</ins></a>
+
+NotUS manages the flow of the application. On launch, it will create the necessary components, as listed above and then attempts to load any existing saved files into the application. Subsequently, it will accept and interpret the user input and execute the commands accordingly. The diagram below depicts the main flow of the application.
+
+<p align="center">
+  <img alt="NotUS" src="diagrams/out/Notus.png" />
+  <br><em>Figure 3</em>
+</p>
+
+#### <a id="parserManager"><ins>2.3 ParserManager</ins></a>
+
+The ParserManager manages the creation of specific parser objects based on the type of command. The parser then makes sense of the user input and calls the respective commands into action. The class diagram is as follows.
+
+<p align="center">
+  <img alt="ParserManagerClass" src="diagrams/out/ParserManagerClass.png" />
+  <br><em>Figure 4</em>
+</p>
```

Comment:
The class diagram is not that clear, you should consider to increase the size of the text in it.